lib/ecto_subscribe/subsribe_api.ex

```diff
@@ -0,0 +1,170 @@
+
+  #
+  # Utils
+  #
+  def query_model_from_system_tbl(repo, model) do
+    #model = Macro.to_string(model)
+    repo.all(from q in Ecto.Subscribe.Schema.SystemTable, where: q.model == ^model)
+  end
+
+  def get_opts_field(opts, field) do
+    case List.keyfind(opts, field, 0) do
+      nil ->
+        ""
+      {_, val} when is_list(val) ->
+        val
+      {_, val} ->
+        val |> to_string
+    end
+  end
+
+  def get_actions("") do
+    [:create, :update, :delete]
+  end
+
+  def get_actions(actions) do
+    actions
+  end
+
+  def find_subscription([], _, _, _) do
+    {:false, []}
+  end
+
+  def find_subscription([subscription_row_in_db | subscription_in_db], changeset, model, action) do
+    find_subscription(subscription_row_in_db, subscription_in_db, changeset, model, action)
+  end
+
+  def find_subscription(subscription_row_in_db, subscription_in_db, changeset, model, action) do
+    find_subscription_helper(subscription_row_in_db, subscription_in_db, changeset, model, action)
+  end
+  
+  def find_subscription_helper(subscription_row_in_db, subscription_in_db, changeset, model, action) do
+    actions = string_to_actions(subscription_row_in_db.subscription_actions)
+    case Enum.member?(actions, action) do
+      true ->
+        {field_name, operator, val} = db_row_to_map(subscription_row_in_db.subscription_info, model)
+        validate_change(subscription_row_in_db, changeset.changes, field_name, operator, val)
+      false ->
+        is_subscription_for_all(actions, subscription_in_db, changeset, model, action)
+    end
+  end
+
+  def validate_change(subscription_row_in_db, changes, field_name, operator, val) do
+    case Map.get(changes, field_name, 0) do
+      nil ->
+        :false
+      updated_val ->
+        case operator do
+          "==" ->
+            {val == updated_val, subscription_row_in_db}
+          ">" ->
+            {updated_val > val, subscription_row_in_db}
+          "<" ->
+            {updated_val > val, subscription_row_in_db}
+          _ ->
+            {:false, []}
+        end
+    end
+  end
+
+  def is_subscription_for_all([:all], subscription_in_db, _, _, _) do
+    {:true, subscription_in_db}
+  end
+
+  def is_subscription_for_all(_, subscription_in_db, changeset, model, action) do
+    find_subscription(subscription_in_db, changeset, model, action)
+  end
+
+  def db_row_to_map(str, model) do
+    [name, operator, val] = String.split(str, " ")
+    type = model.__schema__(:field, name |> String.to_atom)
+    {String.to_atom(name), operator, get_val_with_correct_type_from_string(type, val)}
+  end
+
+  def actions_to_string(actions) do
+    Enum.reduce(actions, "", fn(act, acc) -> acc <> (act |> Atom.to_string) <> "," end) |> String.strip ?,
+  end
+
+  def string_to_actions(str) do
+    splitten_string = String.split(str, ",")
+    for s <- splitten_string do
+      s |> String.to_atom
+    end
+  end
+
+  # TODO think about more flexible way how to convert to correct types
```

Comment:
Ecto.Changeset.cast can do it, but it can be done in a next iteration (I'll tried it exd for input and update calls).